utils.bal

```diff
@@ -0,0 +1,723 @@
+    }
+}
+
+# Send DELETE request.
+# 
+# + httpClient - Drive client
+# + path - DELETE URI path
+# + return - boolean or error if not suceeded, True if Deleted successfully.
+function deleteRequest(http:Client httpClient, string path) returns @tainted boolean|error {
+    var httpResponse = httpClient->delete(<@untainted>path);
+    json resp = check checkAndSetErrors(httpResponse);
+    return true;
+}
+
+# Send POST request with  a Payload.
+# 
+# + httpClient - Drive client
+# + path - POST URI path
+# + jsonPayload - Payload of the request.
+# + return - json or error if not suceeded.
+function sendRequestWithPayload(http:Client httpClient, string path, json jsonPayload = ()) 
+                                    returns @tainted json|error {
+    http:Request httpRequest = new;
+    if (jsonPayload != ()) {
+        httpRequest.setJsonPayload(<@untainted>jsonPayload);
+    }
+    var httpResponse = httpClient->post(<@untainted>path, httpRequest);
+    if (httpResponse is http:Response) {
+        int statusCode = httpResponse.statusCode;
+        json | http:ClientError jsonResponse = httpResponse.getJsonPayload();
+        if (jsonResponse is json) {
+            error? validateStatusCodeRes = validateStatusCode(jsonResponse, statusCode);
+            if (validateStatusCodeRes is error) {
+                return validateStatusCodeRes;
+            }
+            return jsonResponse;
+        } else {
+            return getDriveError(jsonResponse);
+        }
+    } else {
+        return getDriveError(<json|error>httpResponse);
+    }
+}
+
+# Send PATCH request with  a Payload.
+# 
+# + httpClient - Drive client
+# + path - PATCH URI path
+# + jsonPayload - Payload of the request
+# + return - json or error if not suceeded.
+function updateRequestWithPayload(http:Client httpClient, string path, json jsonPayload = ())
+returns @tainted json|error {
+    http:Request httpRequest = new;
+    if (jsonPayload != ()) {
+        httpRequest.setJsonPayload(<@untainted>jsonPayload);
+    }
+    var httpResponse = httpClient->patch(<@untainted>path, httpRequest);
+    if (httpResponse is http:Response) {
+        int statusCode = httpResponse.statusCode;
+        json|http:ClientError jsonResponse = httpResponse.getJsonPayload();
+        if (jsonResponse is json) {
+            error? validateStatusCodeRes = validateStatusCode(jsonResponse, statusCode);
+            if (validateStatusCodeRes is error) {
+                return validateStatusCodeRes;
+            }
+            return jsonResponse;
+        } else {
+            return getDriveError(jsonResponse);
+        }
+    } else {
+        return getDriveError(<json|error>httpResponse);
+    }
+}
+
+# Send POST request with  a Payload.
+# 
+# + httpClient - Drive client
+# + path - POST URI path
+# + jsonPayload - Payload of the request
+# + return - json or error if not suceeded.
+function uploadRequestWithPayload(http:Client httpClient, string path, json jsonPayload = ())
```

Comment:
Do we need default value for jsonPayload ?

Reply:
No. We can eliminate that. I did the changes. Thanks Chuhaashanan